administrative-sdk/connection.js

```diff
@@ -194,33 +194,27 @@ class Connection {
    * @param {string} [auth] The authorization header value to pass along with the request.
    */
   _ajaxGet(url, auth) {
-    var request = new XMLHttpRequest();
-    var response = null;
-    request.open('GET', url);
+    const headers = new Headers();
     if (typeof auth !== 'undefined') {
-      request.setRequestHeader('Authorization', auth);
+      headers.append('Authorization', auth);
     }
-    return new Promise(function(resolve, reject) {
-      request.onload = function() {
-        response = Connection._parseResponse(request.responseText);
-        if (request.status >= 200 && request.status < 300) {
-          resolve(response);
-        } else {
-          reject({
-            status: request.status,
-            errors: response
-          });
-        }
-      };
-
-      request.onerror = function() {
-        reject({
-          status: request.status,
-          errors: response
-        });
-      };
-      request.send();
-    });
+    const options = {
+      method: 'GET',
+      headers: headers
+    };
+    return fetch(url, options)
+      .then(response =>
+        (response.json()
+          .then(data => {
+            if (response.ok) {
+              return data;
+            }
+            throw data;
+          })
+      ))
+      .catch(error => {
+        throw error;
+      });
```

Comment:
This can be simplified by chaining the promises, not nesting them. Also the catch seems to be unnecessary.

```js
return fetch(url, options)
  .then(response => response.json())
  .then(data => {
    if (response.ok) {
      return data;
    }
    throw data;
  });
```